Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp

```diff
@@ -41,6 +351,29 @@ ABenchmarkGymGameMode::ABenchmarkGymGameMode()
 	DropCubeClass = DropCubeClassFinder.Class;
 }
 
+void ABenchmarkGymGameMode::BeginPlay()
+{
+	Super::BeginPlay();
+	SpawnManager = NewObject<USpawnManager>(this);
+
+	if (HasAuthority())
+	{
+		// It's possible for CheckCmdLineParameters may not get called on the game mode auth server if no players are spawned on the game mode auth server.
+		// In case this happens, ensure we still call CheckCmdLineParameters after a delay. The delay ensure worker flags are set.
+		FTimerManager& TimerManager = GetWorld()->GetTimerManager();
```

Comment:
Perhaps we can just call `CheckCmdParameters` here instead?
`GameMode` will spawn NPC when it is ready.
There is no need to wait for players.

Reply:
So the reason I believe this needs to be on timer is so that we give a bit of time for the worker to receive worker flag values.
 CheckCmdParameters will start the logic to create PlayerStarts. To run this logic we need to know the PlayerDensity and ExpectedPlayer count so that we ensure we create clusters with the correct number of players.

I will test locally to see if BeginPlay is late enough so that we have received the worker flags. I remember this was a problem at some time ago... but perhaps not longer...

Reply:
Mmm, this still seem like it is an issue (we have to wait for worker flags to be set).

Reply:
`CheckCmdLineParameters` only effects on the first time. 
Is there any chance when the first time it was called and the worker flags didn't came.
In this case, wait for seconds will not change anything. 

If we won't replace worker flags with command line arguments, perhaps we need find a way to make sure we wait worker flag values before we start to play.

Reply:
The issue this is fixing is that it is possible for CheckCmdLineParameters to never be called on a server (this only really happens in local testing). This can happen when FindPlayerStart is not called due to no attempt to spawn a player being made. 

It appears we always call FindPlayerStart after worker flags are set. This is really by chance and you are right that we should have a better way to deal with this. I think ideally we would cache player spawn requests and then action them once we receive the worker flags. Probably out of scope for this PR. I think I might add callback for when the worker values change though which would be neater.

Reply:
@jeffxu-improbable, I have updated the code to attempt to create player start actors only once updates from worker flags are received. Hopefully this is a little neater.

Reply:
Hi @MatthewSandfordImprobable , sorry about reply late, too busy today. I will check it tomorrow. 
As we changed a lot of uptime game mode, it is better to have a uptime test scenario. I think `gdk-7duptime-zone2-1-manual.yaml` would be enough.